cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -180,6 +180,11 @@ public AppMetadataStore(Table table, CConfiguration cConf) {
     }
   }
 
+  @Nullable
+  public ApplicationMeta getApplication(ApplicationId applicationId) {
+    return getApplication(applicationId.getNamespace(), applicationId.getApplication(), applicationId.getVersion());
```

Comment:
we would want people to use this one and not the other. So it would be better to move the logic into this method and make the other method call this one. Ideally we would remove usages of the other as well, but that can be cleaned up in a follow up.

Reply:
This is too much for this pr because there are so many usages, created https://issues.cask.co/browse/CDAP-13619 for cleanup